common/archiver/s3store/queryParser.go

```diff
@@ -0,0 +1,241 @@
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+//go:generate mockgen -copyright_file ../../../LICENSE -package $GOPACKAGE -source queryParser.go -destination queryParser_mock.go -mock_names Interface=MockQueryParser
+
+package s3store
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/xwb1989/sqlparser"
+
+	"github.com/uber/cadence/common"
+)
+
+type (
+	// QueryParser parses a limited SQL where clause into a struct
+	QueryParser interface {
+		Parse(query string) (*parsedQuery, error)
+	}
+
+	queryParser struct{}
+
+	parsedQuery struct {
+		workflowID      *string
+		runID           *string
+		startTime       *int64
+		closeTime       *int64
+		searchPrecision *string
+		emptyResult     bool
+	}
+)
+
+// All allowed fields for filtering
+const (
+	WorkflowID      = "WorkflowID"
+	RunID           = "RunID"
+	StartTime       = "StartTime"
+	CloseTime       = "CloseTime"
+	SearchPrecision = "SearchPrecision"
+)
+
+// Precision specific values
+const (
+	PrecisionDay    = "Day"
+	PrecisionHour   = "Hour"
+	PrecisionMinute = "Minute"
+	PrecisionSecond = "Second"
+)
+const (
+	queryTemplate         = "select * from dummy where %s"
+	defaultDateTimeFormat = time.RFC3339
+)
+
+// NewQueryParser creates a new query parser for filestore
+func NewQueryParser() QueryParser {
```

Comment:
Can you add some explanation for the query syntax to the `README` file? E.g. Which filter is required? What's the limitation? Supported operators etc.

Reply:
I've started a section. English isn't my first language, will improve on it some more.

Reply:
👍